tests/unit/models/gpflow/test_config.py

```diff
@@ -14,72 +14,39 @@
-from tests.util.models.models import fnc_3x_plus_10
 from trieste.models import TrainableProbabilisticModel
+from trieste.models.config import ModelRegistry
 from trieste.models.gpflow import (
     GaussianProcessRegression,
-    GPflowModelConfig,
     SparseVariational,
     VariationalGaussianProcess,
 )
-from trieste.models.optimizer import Optimizer
+from trieste.models.optimizer import BatchOptimizer, Optimizer
 
 
-def test_gpflow_model_config_raises_not_supported_model_type() -> None:
-    x = tf.constant(np.arange(5).reshape(-1, 1), dtype=gpflow.default_float())
-    y = fnc_3x_plus_10(x)
-    model_specs = {"model": GPMC((x, y), gpflow.kernels.Matern32(), gpflow.likelihoods.Gaussian())}
-
-    with pytest.raises(NotImplementedError):
-        GPflowModelConfig(**model_specs)
-
-
-def test_gpflow_model_config_has_correct_supported_models() -> None:
-
-    x = tf.constant(np.arange(5).reshape(-1, 1), dtype=gpflow.default_float())
-    model_specs = {"model": gpr_model(x, fnc_3x_plus_10(x))}
-    model_config = GPflowModelConfig(**model_specs)
-
-    models_mapping: Dict[Any, Callable[[Any, Optimizer], TrainableProbabilisticModel]] = {
-        GPR: GaussianProcessRegression,
-        SGPR: GaussianProcessRegression,
-        VGP: VariationalGaussianProcess,
-        SVGP: SparseVariational,
-    }
-
-    assert model_config.supported_models() == models_mapping
-
-
-def test_gpflow_model_config_has_correct_default_optimizer() -> None:
-
-    x = tf.constant(np.arange(5).reshape(-1, 1), dtype=gpflow.default_float())
-    model_specs = {"model": gpr_model(x, fnc_3x_plus_10(x))}
-    model_config = GPflowModelConfig(**model_specs)
-
-    default_optimizer = gpflow.optimizers.Scipy
-
-    assert isinstance(model_config.optimizer, default_optimizer)
-
-
-def test_gpflow_model_config_allows_changing_default_optimizer() -> None:
-
-    x = tf.constant(np.arange(5).reshape(-1, 1), dtype=gpflow.default_float())
-    model_specs = {
-        "model": gpr_model(x, fnc_3x_plus_10(x)),
-        "optimizer": tf.optimizers.Adam(),
-    }
-    model_config = GPflowModelConfig(**model_specs)
+@pytest.mark.parametrize(
+    "supported_models",
+    [
+        (GPR, GaussianProcessRegression, Optimizer),
+        (SGPR, GaussianProcessRegression, Optimizer),
+        (VGP, VariationalGaussianProcess, BatchOptimizer),
+        (SVGP, SparseVariational, BatchOptimizer),
+    ],
+)
+def test_supported_gpflow_models_are_correctly_registered(
+    supported_models: Tuple[
+        Type[gpflow.models.GPModel], Type[TrainableProbabilisticModel], Type[Optimizer]
+    ]
+) -> None:
 
-    expected_optimizer = tf.optimizers.Adam
+    model_type, interface, optimizer = supported_models
 
-    assert isinstance(model_config.optimizer, expected_optimizer)
+    assert model_type in ModelRegistry.get_registered_models()
+    assert ModelRegistry.get_interface(model_type) == interface
+    assert ModelRegistry.get_optimizer(model_type) == optimizer
```

Comment:
Any reason not to (also) build a ModelConfig like before and check the types of the actual constructed optimizer?

Reply:
huh, not sure whether this is the right place anymore for that - wrappers providing defaults for the optimizers, so this should probably be done in `test_models.py`, not really related to configs
(before it was related to model config's default for the optimizer, but we don't have that anymore, that's something that wrappers should deal with)